src/analysis/normed_space/star.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2021 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import analysis.normed_space.linear_isometry
+
+/-!
+# Normed star rings and algebras
+
+A normed star ring is a star ring endowed with a norm such that the star operation is isometric.
+
+A C⋆-ring is a normed star ring that verifies the stronger condition `∥x⋆ * x∥ = ∥x∥^2` for all `x`.
+If a C⋆-ring is also a star algebra, then it is a C⋆-algebra.
+
+To get a C⋆-algebra `E` over field `𝕜`, use
+`[normed_field 𝕜] [star_ring 𝕜] [cstar_ring E] [normed_algebra 𝕜 E] [star_module 𝕜 E]`.
+
+-/
+
+local postfix `⋆`:1000 := star
+
+/-- A normed star ring is a star ring endowed with a norm such that `star` is isometric. -/
+class normed_star_ring (E : Type*) [normed_ring E] [star_ring E] :=
+(norm_star : ∀ {x : E}, ∥x⋆∥ = ∥x∥)
+
+export normed_star_ring (norm_star)
+attribute [simp] norm_star
+
+/-- A C*-ring is a normed star ring that satifies the stronger condition `∥x⋆ * x∥ = ∥x∥^2`
+for every `x`. -/
+class cstar_ring (E : Type*) extends normed_ring E, star_ring E :=
```

Comment:
Are you sure you want `extends` here? This prevents you from talking about cstar division algebras.

Reply:
No, I'm not sure at all -- the goal was to simplify a bit the number of typeclasses we have to use to summon a C*-algebra, but I'm happy to remove it if blocks anything.

Reply:
Extending `star_ring` is ok as long as you add `cstar_algebra` that extends both this and star_module.

It's extending `normed_ring` that I'm wary of